### ISSUE TITLE
Add a test to cover closing two sibling panes (near) simultaneously

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#1124]
* [ ] Dependent upon #1042.

I'm adding code in #825 to try and make sure that when a pane is closed nearly synchronously with a sibling, it'll close properly. However, manually testing this is _really_ hard, so I'm filing a follow-up task to make sure that's implemented correctly.

I'll need the test framework stuff I'm working on in 1042 to land first though, so this can be done in a unittest.
